tests/functional/test_upgrade_actions.py

```diff
@@ -233,29 +233,31 @@ def test_check_upgrade_actions_relations(self, sdk_client_fs, folder_dir, file_d
         """
         Test bundle action fails before bundle_switch was performed
         """
+
+        jobs_before = sdk_client_fs.job_list()
         bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, folder_dir, 'old'))
         cluster = bundle.cluster_create('Test Cluster for Upgrade')
         sdk_client_fs.upload_from_fs(get_data_dir(__file__, folder_dir, file_dir))
         with allure.step("Check cluster actions list before update"):
-            actions_before = cluster.action_list().data
+            actions_before = cluster.action_list()
             assert len(actions_before) == 1, "Should be 1 action"
             assert actions_before[0].display_name == "dummy_action", "Should be action 'dummy_action'"
         cluster.upgrade().do().wait()
         cluster.reread()
         with allure.step("Check jobs"):
             jobs = sdk_client_fs.job_list()
-            assert len(jobs) >= 3, "There are should be 3 or more jobs"
+            jobs_expected = 3 + len(jobs_before)
+            assert len(jobs) == jobs_expected, f"There are should be {jobs_expected} jobs"
             assert len(
                 set(['first action after switch', 'switch action', 'first_action'])
-                & set([j.display_name for j in jobs.data])
-            ), "Jobs names differ"
-            assert (
-                len([log for data in jobs.data for log in data.log_files]) >= 6
-            ), "There are should be 6 or more log files"
+                & set([j.display_name for j in jobs])
+            ) == 3, "Jobs names differ"
+            assert len([log for data in jobs for log in data.log_files]) >= 6, "There are should be 6 or more log files"
         with allure.step("Check cluster actions list after update"):
-            actions_after = cluster.action_list().data
-            assert len(actions_after) == 2, "Should be 2 actions"
-            assert [action.display_name for action in actions_after] == [
-                'dummy_action',
-                'restore',
-            ], "Should be actions 'dummy_action' and 'restore'"
+            actions_after = cluster.action_list()
+            assert len(actions_after) == 2 if "success" in folder_dir else 1, "Not all actions avaliable"
+            assert (
+                set([action.display_name for action in actions_after]) == (set(['dummy_action', 'restore'])
```

Comment:
Maybe use set comprehension with `{action.display_name for action in actions_after}` and in similar cases?

Reply:
this block has been changed

Reply:
yep, changed, thnx)